benchmarks/dynamo/check_hf_bert_perf_csv.py

```diff
@@ -16,7 +16,8 @@ def check_hf_bert_perf_csv(filename):
     for _, row in df.iterrows():
         model_name = row["name"]
         speedup = row["speedup"]
-        if speedup < 1.19:
+        threshold = 1.19 if model_name != "hf_Bert" else 1.18
```

Comment:
what motivated the tweak of 1.19 vs 1.18?

Reply:
MLIR branch is slightly slower, but we are willing to accept that...

Reply:
But https://github.com/pytorch/pytorch/blob/e7e51b3a5c1d134bb6334da49cbf89221a7a63a0/.ci/pytorch/test.sh#L343 is the only place this file was used and I believe model_name will always be hf_Bert. 

Could you please point me to the actions that gave < 1.19 speedup?  

I'm just a little concerned lowering the value further will make this smoke test less effective (we just lowered it from 1.2 to 1.19 due to some cudagraph option change): it won't be able to catch small/medium regressions as majority output is around 1.20x. I need to collect more recent runs to confirm this.

Reply:
Below is the recent 100 runs of smoke test performance from main branch commits: 

1.2218 https://github.com/pytorch/pytorch/actions/runs/4159981882/jobs/7196799482 https://ossci-raw-job-status.s3.amazonaws.com/log/11289901248 gh-ci-gcp-a100-1
1.2229 https://github.com/pytorch/pytorch/actions/runs/4158514197/jobs/7195979563 https://ossci-raw-job-status.s3.amazonaws.com/log/11288895075 gh-ci-gcp-a100-6
1.2135 https://github.com/pytorch/pytorch/actions/runs/4159133882/jobs/7195085843 https://ossci-raw-job-status.s3.amazonaws.com/log/11287814574 gh-ci-gcp-a100-1
1.2085 https://github.com/pytorch/pytorch/actions/runs/4158308247/jobs/7193733971 https://ossci-raw-job-status.s3.amazonaws.com/log/11286173534 gh-ci-gcp-a100-2
1.2162 https://github.com/pytorch/pytorch/actions/runs/4158166654/jobs/7193564051 https://ossci-raw-job-status.s3.amazonaws.com/log/11285972682 gh-ci-gcp-a100-1
1.2085 https://github.com/pytorch/pytorch/actions/runs/4157868692/jobs/7192800351 https://ossci-raw-job-status.s3.amazonaws.com/log/11285071492 gh-ci-gcp-a100-5
1.2185 https://github.com/pytorch/pytorch/actions/runs/4157496806/jobs/7192154507 https://ossci-raw-job-status.s3.amazonaws.com/log/11284315510 gh-ci-gcp-a100-5
1.213 https://github.com/pytorch/pytorch/actions/runs/4157381540/jobs/7191988162 https://ossci-raw-job-status.s3.amazonaws.com/log/11284115418 gh-ci-gcp-a100-1
1.2215 https://github.com/pytorch/pytorch/actions/runs/4157140020/jobs/7191585322 https://ossci-raw-job-status.s3.amazonaws.com/log/11283643263 gh-ci-gcp-a100-3-iso
1.2182 https://github.com/pytorch/pytorch/actions/runs/4156273731/jobs/7190117191 https://ossci-raw-job-status.s3.amazonaws.com/log/11281901215 gh-ci-gcp-a100-4
1.2147 https://github.com/pytorch/pytorch/actions/runs/4155952301/jobs/7189533003 https://ossci-raw-job-status.s3.amazonaws.com/log/11281195614 gh-ci-gcp-a100-4
1.2128 https://github.com/pytorch/pytorch/actions/runs/4155521125/jobs/7188801476 https://ossci-raw-job-status.s3.amazonaws.com/log/11280342601 gh-ci-gcp-a100-3-iso
1.2174 https://github.com/pytorch/pytorch/actions/runs/4155424903/jobs/7188651161 https://ossci-raw-job-status.s3.amazonaws.com/log/11280166240 gh-ci-gcp-a100-4
1.225 https://github.com/pytorch/pytorch/actions/runs/4155126815/jobs/7188112519 https://ossci-raw-job-status.s3.amazonaws.com/log/11279542392 gh-ci-gcp-a100-4
1.2145 https://github.com/pytorch/pytorch/actions/runs/4154897586/jobs/7187695166 https://ossci-raw-job-status.s3.amazonaws.com/log/11279064950 gh-ci-gcp-a100-1
1.2156 https://github.com/pytorch/pytorch/actions/runs/4153968353/jobs/7186130784 https://ossci-raw-job-status.s3.amazonaws.com/log/11277259906 gh-ci-gcp-a100-1
1.2159 https://github.com/pytorch/pytorch/actions/runs/4153955863/jobs/7186111738 https://ossci-raw-job-status.s3.amazonaws.com/log/11277238661 gh-ci-gcp-a100-2
1.2115 https://github.com/pytorch/pytorch/actions/runs/4153749030/jobs/7185757118 https://ossci-raw-job-status.s3.amazonaws.com/log/11276835866 gh-ci-gcp-a100-4
1.2145 https://github.com/pytorch/pytorch/actions/runs/4153367838/jobs/7185085627 https://ossci-raw-job-status.s3.amazonaws.com/log/11276047070 gh-ci-gcp-a100-5
1.212 https://github.com/pytorch/pytorch/actions/runs/4153350533/jobs/7185061706 https://ossci-raw-job-status.s3.amazonaws.com/log/11276018755 gh-ci-gcp-a100-3-iso
1.2109 https://github.com/pytorch/pytorch/actions/runs/4153318174/jobs/7185013038 https://ossci-raw-job-status.s3.amazonaws.com/log/11275961772 gh-ci-gcp-a100-6
1.2145 https://github.com/pytorch/pytorch/actions/runs/4153159623/jobs/7184744478 https://ossci-raw-job-status.s3.amazonaws.com/log/11275639154 gh-ci-gcp-a100-4
1.2136 https://github.com/pytorch/pytorch/actions/runs/4152879775/jobs/7184270852 https://ossci-raw-job-status.s3.amazonaws.com/log/11275085927 gh-ci-gcp-a100-5
1.226 https://github.com/pytorch/pytorch/actions/runs/4152684130/jobs/7183941753 https://ossci-raw-job-status.s3.amazonaws.com/log/11274687477 gh-ci-gcp-a100-3-iso
1.2082 https://github.com/pytorch/pytorch/actions/runs/4152568251/jobs/7183770652 https://ossci-raw-job-status.s3.amazonaws.com/log/11274469718 gh-ci-gcp-a100-6
1.213 https://github.com/pytorch/pytorch/actions/runs/4152013177/jobs/7182796495 https://ossci-raw-job-status.s3.amazonaws.com/log/11273270573 gh-ci-gcp-a100-6
1.2215 https://github.com/pytorch/pytorch/actions/runs/4150625799/jobs/7180554343 https://ossci-raw-job-status.s3.amazonaws.com/log/11270645429 gh-ci-gcp-a100-1
1.2167 https://github.com/pytorch/pytorch/actions/runs/4150334540/jobs/7180034407 https://ossci-raw-job-status.s3.amazonaws.com/log/11270034460 gh-ci-gcp-a100-1
1.2119 https://github.com/pytorch/pytorch/actions/runs/4149922937/jobs/7179336464 https://ossci-raw-job-status.s3.amazonaws.com/log/11269215697 gh-ci-gcp-a100-4
1.2141 https://github.com/pytorch/pytorch/actions/runs/4149851954/jobs/7179239636 https://ossci-raw-job-status.s3.amazonaws.com/log/11269099838 gh-ci-gcp-a100-1
1.219 https://github.com/pytorch/pytorch/actions/runs/4149490515/jobs/7178613161 https://ossci-raw-job-status.s3.amazonaws.com/log/11268333806 gh-ci-gcp-a100-5
1.2257 https://github.com/pytorch/pytorch/actions/runs/4149449716/jobs/7178556010 https://ossci-raw-job-status.s3.amazonaws.com/log/11268264306 gh-ci-gcp-a100-2
1.2167 https://github.com/pytorch/pytorch/actions/runs/4149390925/jobs/7178465387 https://ossci-raw-job-status.s3.amazonaws.com/log/11268157237 gh-ci-gcp-a100-1
1.2143 https://github.com/pytorch/pytorch/actions/runs/4149231276/jobs/7178192558 https://ossci-raw-job-status.s3.amazonaws.com/log/11267828679 gh-ci-gcp-a100-5
1.2139 https://github.com/pytorch/pytorch/actions/runs/4149154553/jobs/7178024253 https://ossci-raw-job-status.s3.amazonaws.com/log/11267630749 gh-ci-gcp-a100-1
1.2154 https://github.com/pytorch/pytorch/actions/runs/4149116790/jobs/7177928892 https://ossci-raw-job-status.s3.amazonaws.com/log/11267515028 gh-ci-gcp-a100-4
1.2232 https://github.com/pytorch/pytorch/actions/runs/4149092904/jobs/7177882839 https://ossci-raw-job-status.s3.amazonaws.com/log/11267455139 gh-ci-gcp-a100-2
1.2141 https://github.com/pytorch/pytorch/actions/runs/4149034897/jobs/7177815894 https://ossci-raw-job-status.s3.amazonaws.com/log/11267369593 gh-ci-gcp-a100-1
1.2192 https://github.com/pytorch/pytorch/actions/runs/4148768206/jobs/7177310323 https://ossci-raw-job-status.s3.amazonaws.com/log/11266730394 gh-ci-gcp-a100-5
1.21 https://github.com/pytorch/pytorch/actions/runs/4148722646/jobs/7177239780 https://ossci-raw-job-status.s3.amazonaws.com/log/11266640995 gh-ci-gcp-a100-6
1.2163 https://github.com/pytorch/pytorch/actions/runs/4148711312/jobs/7177222674 https://ossci-raw-job-status.s3.amazonaws.com/log/11266619196 gh-ci-gcp-a100-3-iso
1.2234 https://github.com/pytorch/pytorch/actions/runs/4148494185/jobs/7176954247 https://ossci-raw-job-status.s3.amazonaws.com/log/11266283495 gh-ci-gcp-a100-2
1.2182 https://github.com/pytorch/pytorch/actions/runs/4148547028/jobs/7176952535 https://ossci-raw-job-status.s3.amazonaws.com/log/11266281473 gh-ci-gcp-a100-4
1.2167 https://github.com/pytorch/pytorch/actions/runs/4148541657/jobs/7176951974 https://ossci-raw-job-status.s3.amazonaws.com/log/11266280829 gh-ci-gcp-a100-1
1.2111 https://github.com/pytorch/pytorch/actions/runs/4148546633/jobs/7176950471 https://ossci-raw-job-status.s3.amazonaws.com/log/11266279077 gh-ci-gcp-a100-6
1.2267 https://github.com/pytorch/pytorch/actions/runs/4148388094/jobs/7176556717 https://ossci-raw-job-status.s3.amazonaws.com/log/11265750444 gh-ci-gcp-a100-2
1.2153 https://github.com/pytorch/pytorch/actions/runs/4148255193/jobs/7176309215 https://ossci-raw-job-status.s3.amazonaws.com/log/11265422171 gh-ci-gcp-a100-4
1.2192 https://github.com/pytorch/pytorch/actions/runs/4148205545/jobs/7176209680 https://ossci-raw-job-status.s3.amazonaws.com/log/11265292803 gh-ci-gcp-a100-1
1.2217 https://github.com/pytorch/pytorch/actions/runs/4148095404/jobs/7176004069 https://ossci-raw-job-status.s3.amazonaws.com/log/11265013669 gh-ci-gcp-a100-4
1.2156 https://github.com/pytorch/pytorch/actions/runs/4148022057/jobs/7175866395 https://ossci-raw-job-status.s3.amazonaws.com/log/11264825050 gh-ci-gcp-a100-1
1.2254 https://github.com/pytorch/pytorch/actions/runs/4147985817/jobs/7175816656 https://ossci-raw-job-status.s3.amazonaws.com/log/11264757105 gh-ci-gcp-a100-2
1.2184 https://github.com/pytorch/pytorch/actions/runs/4147863696/jobs/7175509691 https://ossci-raw-job-status.s3.amazonaws.com/log/11264355498 gh-ci-gcp-a100-1
1.2214 https://github.com/pytorch/pytorch/actions/runs/4147739977/jobs/7175259385 https://ossci-raw-job-status.s3.amazonaws.com/log/11264018073 gh-ci-gcp-a100-4
1.2202 https://github.com/pytorch/pytorch/actions/runs/4147539627/jobs/7174845389 https://ossci-raw-job-status.s3.amazonaws.com/log/11263464471 gh-ci-gcp-a100-5
1.2163 https://github.com/pytorch/pytorch/actions/runs/4147502289/jobs/7174758096 https://ossci-raw-job-status.s3.amazonaws.com/log/11263350531 gh-ci-gcp-a100-1
1.2162 https://github.com/pytorch/pytorch/actions/runs/4147140162/jobs/7173972386 https://ossci-raw-job-status.s3.amazonaws.com/log/11262287181 gh-ci-gcp-a100-4
1.2175 https://github.com/pytorch/pytorch/actions/runs/4147136670/jobs/7173958673 https://ossci-raw-job-status.s3.amazonaws.com/log/11262269633 gh-ci-gcp-a100-3-iso
1.2151 https://github.com/pytorch/pytorch/actions/runs/4147007443/jobs/7173684530 https://ossci-raw-job-status.s3.amazonaws.com/log/11261903110 gh-ci-gcp-a100-5
1.2143 https://github.com/pytorch/pytorch/actions/runs/4146744343/jobs/7173153817 https://ossci-raw-job-status.s3.amazonaws.com/log/11261188609 gh-ci-gcp-a100-6
1.2266 https://github.com/pytorch/pytorch/actions/runs/4146616319/jobs/7172861381 https://ossci-raw-job-status.s3.amazonaws.com/log/11260799867 gh-ci-gcp-a100-2
1.2255 https://github.com/pytorch/pytorch/actions/runs/4146531041/jobs/7172649043 https://ossci-raw-job-status.s3.amazonaws.com/log/11260516012 gh-ci-gcp-a100-2
1.2169 https://github.com/pytorch/pytorch/actions/runs/4146522435/jobs/7172644567 https://ossci-raw-job-status.s3.amazonaws.com/log/11260509738 gh-ci-gcp-a100-4
1.2267 https://github.com/pytorch/pytorch/actions/runs/4146468152/jobs/7172560140 https://ossci-raw-job-status.s3.amazonaws.com/log/11260393831 gh-ci-gcp-a100-6
1.2177 https://github.com/pytorch/pytorch/actions/runs/4146467627/jobs/7172543039 https://ossci-raw-job-status.s3.amazonaws.com/log/11260371108 gh-ci-gcp-a100-1
1.2262 https://github.com/pytorch/pytorch/actions/runs/4146430719/jobs/7172475172 https://ossci-raw-job-status.s3.amazonaws.com/log/11260279885 gh-ci-gcp-a100-5
1.22 https://github.com/pytorch/pytorch/actions/runs/4146376689/jobs/7172310892 https://ossci-raw-job-status.s3.amazonaws.com/log/11260060547 gh-ci-gcp-a100-3-iso
1.2246 https://github.com/pytorch/pytorch/actions/runs/4146340910/jobs/7172241814 https://ossci-raw-job-status.s3.amazonaws.com/log/11259968821 gh-ci-gcp-a100-4
1.2167 https://github.com/pytorch/pytorch/actions/runs/4146318990/jobs/7172166876 https://ossci-raw-job-status.s3.amazonaws.com/log/11259870843 gh-ci-gcp-a100-3-iso
1.2186 https://github.com/pytorch/pytorch/actions/runs/4146301939/jobs/7172147969 https://ossci-raw-job-status.s3.amazonaws.com/log/11259845618 gh-ci-gcp-a100-1
1.2118 https://github.com/pytorch/pytorch/actions/runs/4146145364/jobs/7171993609 https://ossci-raw-job-status.s3.amazonaws.com/log/11259631162 gh-ci-gcp-a100-5
1.2291 https://github.com/pytorch/pytorch/actions/runs/4146112250/jobs/7171992052 https://ossci-raw-job-status.s3.amazonaws.com/log/11259628982 gh-ci-gcp-a100-2
1.2151 https://github.com/pytorch/pytorch/actions/runs/4146005053/jobs/7171526555 https://ossci-raw-job-status.s3.amazonaws.com/log/11258988152 gh-ci-gcp-a100-1
1.2136 https://github.com/pytorch/pytorch/actions/runs/4145533497/jobs/7171114367 https://ossci-raw-job-status.s3.amazonaws.com/log/11258440258 gh-ci-gcp-a100-5
1.214 https://github.com/pytorch/pytorch/actions/runs/4145518927/jobs/7171081583 https://ossci-raw-job-status.s3.amazonaws.com/log/11258396469 gh-ci-gcp-a100-3-iso
1.21 https://github.com/pytorch/pytorch/actions/runs/4145427418/jobs/7170178245 https://ossci-raw-job-status.s3.amazonaws.com/log/11257179261 gh-ci-gcp-a100-6
1.2229 https://github.com/pytorch/pytorch/actions/runs/4145255170/jobs/7169840961 https://ossci-raw-job-status.s3.amazonaws.com/log/11256725965 gh-ci-gcp-a100-4
1.2206 https://github.com/pytorch/pytorch/actions/runs/4145261888/jobs/7169838579 https://ossci-raw-job-status.s3.amazonaws.com/log/11256722829 gh-ci-gcp-a100-3-iso
1.2184 https://github.com/pytorch/pytorch/actions/runs/4145228103/jobs/7169811191 https://ossci-raw-job-status.s3.amazonaws.com/log/11256685446 gh-ci-gcp-a100-1
1.2216 https://github.com/pytorch/pytorch/actions/runs/4143990423/jobs/7166906923 https://ossci-raw-job-status.s3.amazonaws.com/log/11252827040 gh-ci-gcp-a100-5
1.202 https://github.com/pytorch/pytorch/actions/runs/4143485745/jobs/7166458857 https://ossci-raw-job-status.s3.amazonaws.com/log/11252245814 gh-ci-gcp-a100-4
1.2073 https://github.com/pytorch/pytorch/actions/runs/4143416273/jobs/7165553340 https://ossci-raw-job-status.s3.amazonaws.com/log/11251070655 gh-ci-gcp-a100-2
1.2035 https://github.com/pytorch/pytorch/actions/runs/4142386271/jobs/7163379557 https://ossci-raw-job-status.s3.amazonaws.com/log/11248226827 gh-ci-gcp-a100-1
1.1986 https://github.com/pytorch/pytorch/actions/runs/4142078447/jobs/7162667061 https://ossci-raw-job-status.s3.amazonaws.com/log/11247308832 gh-ci-gcp-a100-4
1.2003 https://github.com/pytorch/pytorch/actions/runs/4141817879/jobs/7162025103 https://ossci-raw-job-status.s3.amazonaws.com/log/11246495529 gh-ci-gcp-a100-5
1.199 https://github.com/pytorch/pytorch/actions/runs/4141810090/jobs/7162009288 https://ossci-raw-job-status.s3.amazonaws.com/log/11246475629 gh-ci-gcp-a100-4
1.1993 https://github.com/pytorch/pytorch/actions/runs/4141708910/jobs/7161818033 https://ossci-raw-job-status.s3.amazonaws.com/log/11246230752 gh-ci-gcp-a100-1
1.2028 https://github.com/pytorch/pytorch/actions/runs/4141263625/jobs/7160917032 https://ossci-raw-job-status.s3.amazonaws.com/log/11245104504 gh-ci-gcp-a100-1
1.2059 https://github.com/pytorch/pytorch/actions/runs/4141030786/jobs/7160468804 https://ossci-raw-job-status.s3.amazonaws.com/log/11244544923 gh-ci-gcp-a100-4
1.2019 https://github.com/pytorch/pytorch/actions/runs/4140908306/jobs/7160204256 https://ossci-raw-job-status.s3.amazonaws.com/log/11244216393 gh-ci-gcp-a100-5
1.2081 https://github.com/pytorch/pytorch/actions/runs/4140891763/jobs/7160167578 https://ossci-raw-job-status.s3.amazonaws.com/log/11244169222 gh-ci-gcp-a100-2
1.2012 https://github.com/pytorch/pytorch/actions/runs/4140856226/jobs/7160106693 https://ossci-raw-job-status.s3.amazonaws.com/log/11244093555 gh-ci-gcp-a100-4
1.1986 https://github.com/pytorch/pytorch/actions/runs/4140476489/jobs/7159393508 https://ossci-raw-job-status.s3.amazonaws.com/log/11243192217 gh-ci-gcp-a100-6
1.198 https://github.com/pytorch/pytorch/actions/runs/4140126546/jobs/7158689262 https://ossci-raw-job-status.s3.amazonaws.com/log/11242312130 gh-ci-gcp-a100-1
1.2091 https://github.com/pytorch/pytorch/actions/runs/4140094186/jobs/7158620754 https://ossci-raw-job-status.s3.amazonaws.com/log/11242223606 gh-ci-gcp-a100-2
1.1963 https://github.com/pytorch/pytorch/actions/runs/4140049904/jobs/7158543283 https://ossci-raw-job-status.s3.amazonaws.com/log/11242124665 gh-ci-gcp-a100-6
1.2044 https://github.com/pytorch/pytorch/actions/runs/4140031306/jobs/7158502638 https://ossci-raw-job-status.s3.amazonaws.com/log/11242072453 gh-ci-gcp-a100-3-iso
1.2086 https://github.com/pytorch/pytorch/actions/runs/4139942806/jobs/7158349241 https://ossci-raw-job-status.s3.amazonaws.com/log/11241876805 gh-ci-gcp-a100-2
1.1988 https://github.com/pytorch/pytorch/actions/runs/4139832120/jobs/7158108547 https://ossci-raw-job-status.s3.amazonaws.com/log/11241578728 gh-ci-gcp-a100-6
1.2072 https://github.com/pytorch/pytorch/actions/runs/4139813286/jobs/7158069763 https://ossci-raw-job-status.s3.amazonaws.com/log/11241529813 gh-ci-gcp-a100-3-iso
1.2041 https://github.com/pytorch/pytorch/actions/runs/4139786730/jobs/7158017692 https://ossci-raw-job-status.s3.amazonaws.com/log/11241464223 gh-ci-gcp-a100-4

max=1.2291x, min=1.1963x.  My recommended threshold should stay at  1.19x.

Reply:
Ok, https://github.com/pytorch/pytorch/actions/runs/4160130336/jobs/7197026851#step:11:1545 this even got lower than 1.18x...

Reply:
All previous runs probably used the same triton version, but this one changes it, and I was told that it's expected to be slower.

Reply:
And the last run is even slower then before, see https://github.com/pytorch/pytorch/actions/runs/4160130336/jobs/7197026851

Reply:
Gotcha. Looks like this is a regression that we expect.

Reply:
> And the last run is even slower then before, see https://github.com/pytorch/pytorch/actions/runs/4160130336/jobs/7197026851

Yep, I was putting the same link above. Looks like you need to put it down even further?  The historical run's variance was 1.2291/1.1963 =  1.0274x variance. you might want to change the value to 1.16x or even 1.15x.

Reply:
> Gotcha. Looks like this is a regression that we expect.

Though I'm interested to know what value does the new Triton commit bring if it even regresses the one "chosen" sentinel model. Does other models benefit more from this commit than hf_Bert?